tests/fixtures/new_fixtures.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2021 MosaicML. All Rights Reserved.
+
+"""These fixtures are shared globally across the test suite."""
+
+import datetime
+import os
+from unittest.mock import Mock
+
+import pytest
+import torch.distributed as dist
+
+from composer.cli.launcher import get_free_tcp_port
+from composer.core import Logger, State
+from composer.core.types import Precision
+
+
+@pytest.fixture
+def minimal_state():
+    """Most minimally defined state possible.
+
+    Tests should configure the state for their specific needs.
+    """
+    return State(model=Mock(),
+                 precision=Precision.FP32,
+                 grad_accum=1,
+                 train_dataloader=Mock(__len__=lambda x: 100),
+                 evaluators=Mock(),
+                 max_duration='100ep')
+
+
+@pytest.fixture
+def empty_logger(minimal_state: State) -> Logger:
+    """Logger without any output configured."""
+    return Logger(state=minimal_state, backends=[])
+
+
+@pytest.fixture
+def init_process_group(monkeypatch):
```

Comment:
I beleive that we already configure dist globally -- see https://github.com/mosaicml/composer/blob/85b60cddfb722e0ecda93464afecf704673dc254/tests/fixtures/distributed_fixtures.py#L17. Since the fixture has autouse, it is called before any test runs. Curious which tests needed this?

Reply:
Yeah I want to eventually move toward removing this `autouse`, since the Trainer itself already initializes the `dist`. For more isolated tests, better to specifically request the fixture when needed, otherwise we can have subtle bugs.

If this hypothesis doesn't work as we port more tests over, we can move this fixture over.

Reply:
Hmm ok, I am now forgetting why we are configuring dist globally. It might not be needed any longer. Testing right now to see if we can rip out the autoused fixture...

Reply:
Just tested it and I think we can rip out `distributed_fixtures.py`. I replaced it with something like the following, and this works for me:

```python3
# Copyright 2021 MosaicML. All Rights Reserved.

import pytest

from composer.utils import dist


@pytest.fixture(autouse=True)
def wait_for_all_procs():
    yield
    dist.barrier()
```